### DISCUSSION TITLE
Silently open a new tab

### DISCUSSION THREAD

**Discussion by TroySigX** [nvim-tree/nvim-tree.lua#1875] [Show and tell]
Please add an option to open a tab (when pressing <C-t>) silently. That is, staying on the current buffer rather than jumping to new tab's buffer.

Suggestion: create a new mapping <Shift-t> to this new feature.

**Comment by alex-courtis** [comment]
To confirm:
* user presses, say, `ts` on a file
* new tab is opened
* new tab contains one window with a new buffer with the file
* new tab is not visible as the user stays on the current tab
* tree window remains focused

You should be able to do that yourself with a custom action, see `:help nvim-tree-mappings` and user's [Recipes](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes)

If you can't please reopen / comment.

**Comment by TroySigX** [comment]
I'm pretty new to lua. Can you show me the code?

**Comment by alex-courtis** [comment]
> I'm pretty new to lua. Can you show me the code?

Take a look at the recipes linked above. [Fix tab titles when opening file in new tab](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#fix-tab-titles-when-opening-file-in-new-tab) might be a good start.

**Comment by TroySigX** [comment]
Thank you very much! I will do that. But I do believe that it will be much helpful if this mapping is in the official code.

**Comment by alex-courtis** [comment]
> Thank you very much! I will do that. But I do believe that it will be much helpful if this mapping is in the official code.

We are no longer adding specific new functionality: https://github.com/nvim-tree/nvim-tree.lua#roadmap

nvim-tree has suffered from feature creep and will only add QOL features that cannot be achieved via API.

**Comment by TroySigX** [comment]
If not, can you please provide the config? I believe this feature is super useful for many users.

**Comment by alex-courtis** [comment]
Here's a quick implementation. You might need to tweak it a bit to better suit your needs.

```lua
local function open_tab_silent(node)
  local api = require("nvim-tree.api")

  api.node.open.tab(node)

  vim.cmd.tabprev()
end
---
      mappings = {
        list = {
          { key = "ts", action = "open_tab_silent", action_cb = open_tab_silent },
```

**Comment by TroySigX** [comment]
Thank you so much!

**Comment by alex-courtis** [comment]
Glad to hear this worked.

Could you please add your final solution to the [Recipes](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes) page?

I don't think the draft works very well around multiple tabs...
